cpp/oneapi/dal/detail/communicator.cpp

```diff
@@ -52,15 +53,22 @@ spmd::request_iface* spmd_communicator_via_host_impl::bcast(sycl::queue& q,
     const std::int64_t dtype_size = get_data_type_size(dtype);
     const std::int64_t size = check_mul_overflow(dtype_size, count);
 
-    const auto send_buff_host = array<byte_t>::empty(size);
-    if (get_rank() == root) {
-        memcpy_usm2host(q, send_buff_host.get_mutable_data(), send_buf, size);
+    if (get_mpi_gpu_support()) {
```

Comment:
does the adding of `const bool gpu_offloading = get_mpi_gpu_support() ` and after that reuse this variable instead of calling the function make sense?